protocol/validator.go

```diff
@@ -0,0 +1,55 @@
+package protocol
+
+import (
+	"fmt"
+
+	"github.com/TBD54566975/tbdex-go"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+// ValidatorMap contains a map of json schema name to schema validator
+var ValidatorMap map[string]*jsonschema.Schema = make(map[string]*jsonschema.Schema)
+
+type schemaObject struct {
+	URL  string
+	Path string
+}
+
+var schemas = map[string]schemaObject{
+	"definitions": {"https://tbdex.dev/definitions.json", "tbdex/hosted/json-schemas/definitions.json"},
+	"resource":    {"https://tbdex.dev/resource.schema.json", "tbdex/hosted/json-schemas/resource.schema.json"},
+	"offering":    {"https://tbdex.dev/offering.schema.json", "tbdex/hosted/json-schemas/offering.schema.json"},
+	"balance":     {"https://tbdex.dev/balance.schema.json", "tbdex/hosted/json-schemas/balance.schema.json"},
+	"message":     {"https://tbdex.dev/message.schema.json", "tbdex/hosted/json-schemas/message.schema.json"},
+	"order":       {"https://tbdex.dev/order.schema.json", "tbdex/hosted/json-schemas/order.schema.json"},
+	"orderstatus": {"https://tbdex.dev/orderstatus.schema.json", "tbdex/hosted/json-schemas/orderstatus.schema.json"},
+	"quote":       {"https://tbdex.dev/quote.schema.json", "tbdex/hosted/json-schemas/quote.schema.json"},
+	"rfq":         {"https://tbdex.dev/rfq.schema.json", "tbdex/hosted/json-schemas/rfq.schema.json"},
+	"close":       {"https://tbdex.dev/close.schema.json", "tbdex/hosted/json-schemas/close.schema.json"},
```

Comment:
The path seems redundant here as they're all the same? You could do `filepath.Join("tbdex/hosted/json-schemas", path.Base("https://tbdex.dev/close.schema.json"))`, as it would make maintenance a bit simpler.